plugins/store/src/store.rs

```diff
@@ -290,6 +307,69 @@ impl<R: Runtime> Store<R> {
     pub fn is_empty(&self) -> bool {
         self.cache.is_empty()
     }
+
+    fn emit_change_event(&self, key: &str, value: &JsonValue) -> Result<(), Error> {
+        self.app.emit(
+            "store://change",
+            ChangePayload {
+                path: &self.path,
+                key,
+                value,
+            },
+        )?;
+        Ok(())
+    }
+
+    fn trigger_auto_save(&mut self) -> Result<(), Error> {
+        let Some(auto_save_delay) = self.auto_save else {
+            return Ok(());
+        };
+        if auto_save_delay.is_zero() {
+            return self.save();
+        }
+        if let Some(sender) = &self.auto_save_debounce_sender {
+            let _ = sender.send(AutoSaveMessage::Reset);
+            return Ok(());
+        }
+        let (sender, mut receiver) = unbounded_channel();
+        self.auto_save_debounce_sender.replace(sender);
+        let app = self.app.clone();
+        let path = self.path.clone();
+        tauri::async_runtime::spawn(async move {
+            loop {
+                select! {
+                    should_cancel = receiver.recv() => {
+                        if matches!(should_cancel, Some(AutoSaveMessage::Cancel) | None) {
+                            return;
+                        }
+                    }
+                    _ = sleep(auto_save_delay) => {
+                        let collection = app.state::<StoreCollection<R>>();
+                        if let Some(store) = collection
+                            .stores
+                            .lock()
+                            .expect("mutex poisoned")
+                            .values_mut()
+                            .find(|store| store.path == path)
+                        {
+                            let _ = store.save();
+                            store.auto_save_debounce_sender = None;
```

Comment:
This right now only works if the store is managed by `app.state`, if we want it to work with plain `Store` we probably need to wrap the inner in an `Arc<Mutex>` which I don't know if it's worth it